Build failure when compiling against libjpeg-turbo 2.0

After upgrading from libjpeg-turbo 1.5.3 to 2.0.0 from https://github.com/libjpeg-turbo/libjpeg-turbo/releases/tag/2.0.0, libheif fails to build:

```
(min) hmage@min:~/norm.x86_64-pc-linux-gnu.2.24.haswell.min/tmp/compile/libheif/libheif_libheif-1.3.2.tar.gz$ make
make  all-recursive
make[1]: Entering directory '/home/hmage/norm.x86_64-pc-linux-gnu.2.24.haswell.min/tmp/compile/libheif/libheif_libheif-1.3.2.tar.gz'
Making all in libheif
make[2]: Entering directory '/home/hmage/norm.x86_64-pc-linux-gnu.2.24.haswell.min/tmp/compile/libheif/libheif_libheif-1.3.2.tar.gz/libheif'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/hmage/norm.x86_64-pc-linux-gnu.2.24.haswell.min/tmp/compile/libheif/libheif_libheif-1.3.2.tar.gz/libheif'
Making all in examples
make[2]: Entering directory '/home/hmage/norm.x86_64-pc-linux-gnu.2.24.haswell.min/tmp/compile/libheif/libheif_libheif-1.3.2.tar.gz/examples'
c++ -DHAVE_CONFIG_H -I. -I..   -I/home/hmage/norm.x86_64-pc-linux-gnu.2.24.haswell.min/include -march=native -fdebug-prefix-map=/home/hmage/norm.x86_64-pc-linux-gnu.2.24.haswell.min=/build -ffunction-sections -fdata-sections -I.. -I../.   -g -O2 -Wall -Werror -Wsign-compare -Wconversion -Wno-sign-conversion -Wno-error=conversion -c -o heif_convert-heif_convert.o `test -f 'heif_convert.cc' || echo './'`heif_convert.cc
In file included from /home/hmage/norm.x86_64-pc-linux-gnu.2.24.haswell.min/include/jpeglib.h:29:0,
                 from encoder_jpeg.h:27,
                 from heif_convert.cc:38:
/home/hmage/norm.x86_64-pc-linux-gnu.2.24.haswell.min/include/jconfig.h:39:0: error: "HAVE_STDDEF_H" redefined [-Werror]
 #define HAVE_STDDEF_H

In file included from heif_convert.cc:21:0:
../config.h:38:0: note: this is the location of the previous definition
 #define HAVE_STDDEF_H 1

In file included from /home/hmage/norm.x86_64-pc-linux-gnu.2.24.haswell.min/include/jpeglib.h:29:0,
                 from encoder_jpeg.h:27,
                 from heif_convert.cc:38:
/home/hmage/norm.x86_64-pc-linux-gnu.2.24.haswell.min/include/jconfig.h:42:0: error: "HAVE_STDLIB_H" redefined [-Werror]
 #define HAVE_STDLIB_H

In file included from heif_convert.cc:21:0:
../config.h:44:0: note: this is the location of the previous definition
 #define HAVE_STDLIB_H 1

cc1plus: all warnings being treated as errors
Makefile:662: recipe for target 'heif_convert-heif_convert.o' failed
make[2]: *** [heif_convert-heif_convert.o] Error 1
make[2]: Leaving directory '/home/hmage/norm.x86_64-pc-linux-gnu.2.24.haswell.min/tmp/compile/libheif/libheif_libheif-1.3.2.tar.gz/examples'
Makefile:507: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/hmage/norm.x86_64-pc-linux-gnu.2.24.haswell.min/tmp/compile/libheif/libheif_libheif-1.3.2.tar.gz'
Makefile:416: recipe for target 'all' failed
make: *** [all] Error 2
```

Didn't have time yet to track down exactly why, but better have the bugreport rather than not.

Another one with libjpeg-turbo-2.0 but GCC 8.2 on Gentoo:
```
/var/tmp/portage/media-libs/libheif-9999/work/libheif-9999/examples/encoder_jpeg.cc:64:6: error: ‘void jpeg_write_icc_profile(j_compress_ptr, const JOCTET*, unsigned int)’ was declared ‘extern’ and later ‘static’ [-fpermissive]
 void jpeg_write_icc_profile(j_compress_ptr cinfo, const JOCTET *icc_data_ptr,
      ^~~~~~~~~~~~~~~~~~~~~~
In file included from /var/tmp/portage/media-libs/libheif-9999/work/libheif-9999/examples/encoder_jpeg.h:27,
                 from /var/tmp/portage/media-libs/libheif-9999/work/libheif-9999/examples/encoder_jpeg.cc:26:
/usr/include/jpeglib.h:979:14: note: previous declaration of ‘void jpeg_write_icc_profile(j_compress_ptr, const JOCTET*, unsigned int)’
 EXTERN(void) jpeg_write_icc_profile(j_compress_ptr cinfo,
              ^~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:693: heif_convert-encoder_jpeg.o] Error 1
```

--- a/configure.ac
+++ b/configure.ac
@@ -105,6 +105,22 @@ AC_CHECK_LIB([jpeg], [jpeg_destroy_compress], [
 ], [have_libjpeg="no"])
 fi
 AM_CONDITIONAL([HAVE_LIBJPEG], [test "x$have_libjpeg" = "xyes"])
+if eval "test x$have_libjpeg = xyes"; then
+AC_MSG_CHECKING([for jpeg_write_icc_profile])
+AC_LANG_PUSH(C++)
+AC_TRY_COMPILE([
+    #include <stddef.h>
+    #include <stdio.h>
+    #include <jpeglib.h>
+],[
+    jpeg_write_icc_profile(NULL, NULL, 0);
+],[has_jpeg_write_icc_profile=yes],[has_jpeg_write_icc_profile=no]);
+AC_LANG_POP(C++)
+AC_MSG_RESULT([$has_jpeg_write_icc_profile])
+if eval "test x$has_jpeg_write_icc_profile = xyes"; then
+    AC_DEFINE(HAVE_JPEG_WRITE_ICC_PROFILE, 1, [Define to 1 if jpeg_write_icc_profile is available in libjpeg.])
+fi
+fi
 
 PKG_CHECK_MODULES([libpng], [libpng], [
     AC_DEFINE([HAVE_LIBPNG], [1], [Whether libpng was found.])

--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -10,6 +10,24 @@ include (${CMAKE_ROOT}/Modules/FindJPEG.cmake)
 
 if(JPEG_FOUND)
 add_definitions(-DHAVE_LIBJPEG=1)
+
+include (${CMAKE_ROOT}/Modules/CheckCXXSourceCompiles.cmake)
+
+set(CMAKE_REQUIRED_LIBRARIES ${JPEG_LIBRARIES})
+check_cxx_source_compiles("
+#include <stddef.h>
+#include <stdio.h>
+#include <jpeglib.h>
+
+int main() {
+  jpeg_write_icc_profile(NULL, NULL, 0);
+  return 0;
+}
+" HAVE_JPEG_WRITE_ICC_PROFILE)
+if(HAVE_JPEG_WRITE_ICC_PROFILE)
+  add_definitions(-DHAVE_JPEG_WRITE_ICC_PROFILE=1)
+endif()
+
 set (heif_convert_sources
   ${heif_convert_sources}
   encoder_jpeg.cc

--- a/examples/encoder_jpeg.cc
+++ b/examples/encoder_jpeg.cc
@@ -23,6 +23,10 @@
   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
   SOFTWARE.
 */
+#if defined(HAVE_CONFIG_H)
+#include "config.h"
+#endif
+
 #include <assert.h>
 #include <errno.h>
 #include <string.h>
@@ -50,7 +54,7 @@ void JpegEncoder::OnJpegError(j_common_ptr cinfo) {
   longjmp(handler->setjmp_buffer, 1);
 }
 
-
+#if !defined(HAVE_JPEG_WRITE_ICC_PROFILE)
 
 #define ICC_MARKER  (JPEG_APP0 + 2)     /* JPEG marker code for ICC */
 #define ICC_OVERHEAD_LEN  14            /* size of non-profile data in APP2 */
@@ -119,6 +123,8 @@ void jpeg_write_icc_profile(j_compress_ptr cinfo, const JOCTET *icc_data_ptr,
     cur_marker++;
   }
 }
+
+#endif  // !defined(HAVE_JPEG_WRITE_ICC_PROFILE)
 
 bool JpegEncoder::Encode(const struct heif_image_handle* handle,
     const struct heif_image* image, const std::string& filename) {

--- a/examples/encoder_jpeg.h
+++ b/examples/encoder_jpeg.h
@@ -30,6 +30,11 @@
 #include <stddef.h>
 #include <stdio.h>
 
+// Prevent duplicate definition for libjpeg-turbo v2.0
+// Note: these 'undef's are only a workaround for a libjpeg-turbo-v2.0 bug and
+// should be removed again later. Bug has been fixed in libjpeg-turbo-v2.0.1.
+#undef HAVE_STDDEF_H
+#undef HAVE_STDLIB_H
 #include <jpeglib.h>
 
 #include <string>

--- a/examples/heif_enc.cc
+++ b/examples/heif_enc.cc
@@ -41,6 +41,11 @@
 
 #if HAVE_LIBJPEG
 extern "C" {
+// Prevent duplicate definition for libjpeg-turbo v2.0
+// Note: these 'undef's are only a workaround for a libjpeg-turbo-v2.0 bug and
+// should be removed again later. Bug has been fixed in libjpeg-turbo-v2.0.1.
+#undef HAVE_STDDEF_H
+#undef HAVE_STDLIB_H
 #include <jpeglib.h>
 }
 #endif

